src/fsharp/FSharp.Build/Microsoft.FSharp.targets

```diff
@@ -66,7 +66,7 @@ this file.
 
     <PropertyGroup>
         <UsingXBuild>false</UsingXBuild>
-        <UsingXBuild Condition="$(MSBuildExtensionsPath32.Contains('xbuild'))">true</UsingXBuild>
+        <UsingXBuild Condition="'$(MSBuildAssemblyVersion)' == ''">true</UsingXBuild>
```

Comment:
Just for info sake, if you want to differentiate between `msbuild` running on mono/.net/netcore, then you can use `$(MSBuildRuntimeType)` which can have values `Mono`, `Full` or `Core`.